### DISCUSSION TITLE
v1.39.0

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#1581] [General]
## Features
In this release we focused mainly on internal changes to improve security, performance and database support. We've also introduced a new UI  that allows you to query your event data. In the events section, you will see a new button called `Event Data`.

![image](https://user-images.githubusercontent.com/477975/197660814-d6773014-9131-44e3-ba3b-e990cf45fd13.png)

From there you'll be able to query your JSON data by selecting columns and applying filters.

![image](https://user-images.githubusercontent.com/477975/197661217-975b9855-8201-4a4b-8266-cf9830a0738a.png)

## News

This will probably be one of the last v1 releases we do as we gear up for v2 of Umami. In v2 we'll be making changes in that will enable lots of new features, such as:
- Advanced user/role/group based permissions so you can share access to website with your team
- Completely customizable dashboard that will let you show data however you want
- A/B testing
- Visitor funnels
- Session recording

and much more!

A reminder, if you are interested in joining our **free** beta program before we launch, you can sign up at https://umami.is/beta.

## Fixes
- Fixed wrong dot color on real-time page
- Language translation updates

## Updates
- Upgraded Next to `12.3.1`
- Upgraded Prisma to `4.5.0`

A huge thanks to all the contributors on this release! @rebron1900 @franciscao633 @ariaieboy @userdehghani @mikusaa @briancao @bilguun0203 @AliMickey @roywangdev @sergiomensing @SeesawLiu 

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v1.39.0'>v1.39.0</a>.</em>

**Comment by aidanm1999** [comment]
## v1.39.1 - PlanetScale Upgrade Guide

I made this guide to help anyone else that comes into the same issue when updating to `v1.39.1` and applying the `04_add_uuid` migration. If your production branch is not protected, this guide does not apply and you can just run `yarn build` after pulling & installing. For devs that are using a protected branch which you cannot push updates to and instead are using `Deploy Requests` to update your main database branch, do not run `yarn build` after `git pull && yarn install`.

This is because in the 4th migration (`04_add_uuid`), the second command backfills UUID which allows the third command to be run:

```sql
ALTER TABLE account MODIFY account_uuid VARCHAR(36) NOT NULL;
```

If you were to create a deploy request, it would only copy over schema changes instead of all the migration which means this command will not be run, causing the request to fail:

```sql
UPDATE `account` SET account_uuid=(SELECT uuid());
```

Instead, do the following:

1. Comment out lines `8` and `20`
2. Create a Deploy Request from your development branch to your production branch
3. Merge request successfully & don't delete the development branch
4. Uncomment lines `8` and `20`
5. In your protected branch, open the console and run lines `5` and `17` which populates the account uuids
6. In the development branch, run the two commands from lines `8` and `20`
7. Merge request successfully

And that should be it!

**Comment by javimosch** [comment]
Will you update the related docker image with this version?

https://hub.docker.com/layers/umamisoftware/umami/mysql-latest/images/sha256-649c365a4c0b1d2d1bd22f65c4703bb84989de75d855a9d60cb1b7427b8098d5?context=explore

**Comment by psifertex** [comment]
Was this feature removed in 2.0? I don't see it in the documentation or the UI anywhere.

**Reply by psifertex** [reply]
Ahh, maybe event data has to be gathered over time? I do have live raw event data but the actual "event data" section up top is empty.

**Reply by vashchukmaksim** [reply]
Hi, did you find out that functionality in v2? I can't see it either even though we have events collected

**Reply by franciscao633** [reply]
We have an event data section that displays simple summarized data, but the querying mechanism was removed in v2. 

![image](https://github.com/umami-software/umami/assets/31608805/3907c66d-c30b-4597-8393-928d04e2ba89)

We have a refactor of event data with similar functionality in our next release or so.

**Reply by vashchukmaksim** [reply]
Thanks for a quick reply! Yes, we use events and "Event data" section, that is the most important part and it works very well but more flexible visualization of events data and ability to query data by parameters are the parts that we miss the most.

We consider a separate solution in a longer term to connect to Umami database just for these purposes. If there is related functionality expected in the next releases that's really nice 👍
